homeassistant/components/mcp_server/config_flow.py

```diff
@@ -32,19 +32,26 @@ async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Handle the initial step."""
+        errors: dict[str, str] = {}
         llm_apis = {api.id: api.name for api in llm.async_get_apis(self.hass)}
         if user_input is not None:
-            return self.async_create_entry(
-                title=llm_apis[user_input[CONF_LLM_HASS_API]], data=user_input
-            )
+            if not user_input[CONF_LLM_HASS_API]:
+                errors[CONF_LLM_HASS_API] = "llm_api_required"
+            else:
+                return self.async_create_entry(
+                    title=", ".join(
+                        llm_apis[api_id] for api_id in user_input[CONF_LLM_HASS_API]
+                    ),
+                    data=user_input,
+                )
 
         return self.async_show_form(
             step_id="user",
             data_schema=vol.Schema(
                 {
                     vol.Optional(
```

Comment:
Hmm I wonder if this was a mistake and it should really be `vol.Required`?

Reply:
`vol.Required` doesn't protect from the empty list value, I've just tested.